WordPress/gutenberg
# Editor: Refactor `PostPreviewButton` tests to `@testing-library/react`

## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `IsolatedEventContainer` tests from `enzyme` to `@testing-library/react`.

One of the last remaining bits of #17249.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're refactoring the tests a bit to test more the from the user perspective and less from the implementation detail perspective.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/editor/src/components/post-preview-button/test/index.js`
